mdit_py_plugins/dollarmath/index.py

```diff
@@ -302,15 +302,14 @@ def _math_block_dollar(
                 start = state.bMarks[nextLine] + state.tShift[nextLine]
                 end = state.eMarks[nextLine]
 
-                if end - start < 2:
```

Comment:
I think this should stay (but break not continue) to safeguard against any index errors

Reply:
This line doesn't exist in the javascript version, why would you want to have it in the Python version?

Reply:
Because JavaScript does not act the same as Python; if you index a JavaScript array with "bad indexes" it will simply return null, whereas in python it will raise an indexerror.
There has been multiple bug fixed for this in markdown-it-py

Reply:
The only indexing that takes place is `state.src[start:end]`, and that will never give an `IndexError`

Reply:
@rowanc1: what can we do to progress here?